src/parser/deserializer.ts

```diff
@@ -604,12 +603,12 @@ function deserializeObject(
       )
         throw new BSONError('bad string length in bson');
       // Namespace
-      if (validation != null && validation.utf8) {
-        if (!validateUtf8(buffer, index, index + stringSize - 1)) {
-          throw new BSONError('Invalid UTF-8 string in BSON document');
-        }
-      }
-      const namespace = ByteUtils.toUTF8(buffer, index, index + stringSize - 1, false);
+      const namespace = ByteUtils.toUTF8(
+        buffer,
+        index,
+        index + stringSize - 1,
+        validation != null && (validation.utf8 as boolean)
```

Comment:
I think this may be the incorrect way to determine validation, shouldn't this be `shouldValidateKey`?

Reply:
I've made the change and it doesn't affect any tests. 
It does stray from the method used by the rest of this function, but I wonder why this was implemented this way?

Reply:
Yes, I'll add a test, messaged you on Slack for more clarification